src/orion/client/experiment.py

```diff
@@ -33,51 +30,39 @@
 log = logging.getLogger(__name__)
 
 
-def reserve_trial(experiment, producer, pool_size, timeout):
+def reserve_trial(experiment, producer, pool_size, timeout=None):
     """Reserve a new trial, or produce and reserve a trial if none are available."""
     log.debug("Trying to reserve a new trial to evaluate.")
 
-    trial = None
-    start = time.time()
-    failure_count = 0
+    if timeout is not None:
+        log.warning("reservation_timeout is deprecated")
```

Comment:
```suggestion
        log.warning("Reservation_timeout is deprecated and will be removed in v0.4.0. Use idle_timeout instead.")
```
We should already plan for the deprecation time. I would be in favor of v0.4.0.